### PR TITLE
HW-52223 Implement New GraphQl Key Data Structures

### PR DESCRIPTION
**Pull request by syyang-hw**

### CODE REVIEW

File: Sources/Model/Meta/HyperwalletTransferMethodConfigurationField.swift [hunk 1]
```diff
@@ -42,13 +42,12 @@ final class TransferMethodConfigurationFieldResult: HyperwalletTransferMethodCon
     /// - Parameters:
     ///   - transferMethodUIConfigurations: the GraphQL `[TransferMethodUIConfiguration]`
     ///   - transferMethodTypes: the GraphQL `HyperwalletTransferMethodType`
```
Comment (jsaini-hw) [inline Sources/Model/Meta/HyperwalletTransferMethodConfigurationField.swift]: Update the comment

File: Tests/HyperwalletTransferMethodConfigurationTests.swift [hunk 2]
```diff
@@ -1,147 +1,138 @@
...
-//            .buildPostResquest(baseUrl: HyperwalletTestHelper.graphQlURL,
-//                               HyperwalletTestHelper.setUpMockedResponse(payload: data, error: error))
-//    }
-//}
+import Hippolyte
+@testable import HyperwalletSDK
+import XCTest
+
+//swiftlint:disable multiline_arguments
+class HyperwalletTransferMethodConfigurationTests: XCTestCase {
```
Comment (jsaini-hw) [inline Tests/HyperwalletTransferMethodConfigurationTests.swift]: split the test class for keys and fields test cases

File: Tests/HyperwalletTransferMethodConfigurationTests.swift [hunk 3]
```diff
@@ -1,147 +1,138 @@
...
+        XCTAssertEqual(graphQlResponse?.currencies(from: "CA")?.count, 2, "currencies(...)` should be 2")
+        XCTAssertEqual(graphQlResponse?.transferMethodTypes(countryCode: "CA", currencyCode: "CAD")?
+                                       .count, 2, "`transferMethodTypes(...)` should be 2")
+        fees = graphQlResponse?.transferMethodTypes(countryCode: "CA", currencyCode: "CAD")?
+                               .first(where: { $0.code == "BANK_ACCOUNT" })
+                               .flatMap { $0.fees?.nodes }
+        flatFee = fees?.first
+        percentFee = fees?.last
+
+        XCTAssertNotNil(fees)
```
Comment (jsaini-hw) [inline Tests/HyperwalletTransferMethodConfigurationTests.swift]: Add an assert for processingTime as well

File: Tests/HyperwalletTransferMethodConfigurationTests.swift [hunk 4]
```diff
@@ -1,147 +1,138 @@
...
+            })
+
+            wait(for: [expectation], timeout: 1)
+
+            // Then
+            XCTAssertNil(errorResponse, "The `errorResponse` should be nil")
+            XCTAssertNotNil(graphQlResponse)
+            XCTAssertEqual(graphQlResponse?.fieldGroups()?.count, 2, "`fieldGroups()` should be 2")
+            let fees = graphQlResponse?.transferMethodType()?.fees?.nodes
+            XCTAssertNotNil(fees)
```
Comment (jsaini-hw) [inline Tests/HyperwalletTransferMethodConfigurationTests.swift]: Assert for each field

File: Tests/HyperwalletTransferMethodConfigurationTests.swift (hunk 2 above)
Comment (jsaini-hw) [reply Tests/HyperwalletTransferMethodConfigurationTests.swift]: My bad not this class

File: Tests/TransferMethodConfigurationResultTests.swift [hunk 5]
```diff
@@ -1,285 +1,178 @@
...
-//        let graphQlResult = try! decoder.decode(GraphQlResult<Connection<TransferMethodConfiguration>>.self,
-//                                                from: data)
-//        return graphQlResult.data?["transferMethodConfigurations"]
-//    }
-//}
+@testable import HyperwalletSDK
+import XCTest
+
+//swiftlint:disable force_try
+class TransferMethodConfigurationResultTests: XCTestCase {
```
Comment (jsaini-hw) [inline Tests/TransferMethodConfigurationResultTests.swift]: split the class for keys and fields

Comment (syyang-hw) [reply Tests/TransferMethodConfigurationResultTests.swift]: Sure.

File: Sources/Model/Meta/HyperwalletTransferMethodConfigurationField.swift (hunk 1 above)
Comment (syyang-hw) [reply Sources/Model/Meta/HyperwalletTransferMethodConfigurationField.swift]: Sure

File: README.md [hunk 6]
```diff
@@ -358,18 +358,21 @@ Hyperwallet.shared.retrieveTransferMethodConfigurationKeys(request: keysQuery) {
 
     guard let result = result else { return }
     // Get countries
-    let countryCodes = result.countries()
+    let countries = result.countries()
```
Comment (oleksii-lubianyi-epam) [inline README.md]: You already have `countries` on line 365, don't you?

File: README.md [hunk 7]
```diff
@@ -358,18 +358,21 @@ Hyperwallet.shared.retrieveTransferMethodConfigurationKeys(request: keysQuery) {
...
+    if let countries = result.countries(), !countries.isEmpty {
+        currencies = result.currencies(from: countries.first!.code)
+    }
     
-    // Get transfer method types based on the country, currency and profile type
-    let transferMethodTypes = result.transferMethodTypes(country: countryCodes.first,
-                                                         currency: transferMethodCurrencies.first,
-                                                         profileType: "INDIVIDUAL") // Could be INDIVIDUAL or BUSINESS
+    // Get transfer method types based on the first country code and its first currency code
+    if let countryCode = countries?.first?.code, let currencyCode = currencies?.first?.code {
```
Comment (oleksii-lubianyi-epam) [inline README.md]: May be included in previous `if let`

Comment (syyang-hw) [reply README.md]: Sorry, I did not get it.

File: README.md (hunk 6 above)
Comment (syyang-hw) [reply README.md]: Right.

**Pull request merged**